src/views/connectionUI.ts

```diff
@@ -498,40 +503,25 @@ export class ConnectionUI {
      * false otherwise
      */
     public async handleFirewallError(uri: string, profile: IConnectionProfile, ipAddress: string): Promise<boolean> {
-        // Check whether the azure account extension is installed and active
-        if (this._vscodeWrapper.azureAccountExtensionActive) {
-            // Sign in to azure account
-            const signedIn = await this.promptForAccountSignIn();
-            if (signedIn) {
-                // Create a firewall rule for the server
-                let success = await this.createFirewallRule(profile, profile.server, ipAddress);
-                if (success) {
-                    this._vscodeWrapper.showInformationMessage(LocalizedConstants.msgPromptFirewallRuleCreated);
-                }
-                return success;
-            }
+        let azureController = new AzureController(this._context);
+        // TODO: Access account which firewall error needs to be added from:
+        // Try to match accountId to an account in account storage
+        if (profile.accountId) {
+            let account = this._accountStore.getAccount(profile.accountId);
+            this.connectionManager.accountService.setAccount(account);
+            // take that account from account storage and refresh tokens and create firewall rule
         } else {
-            // If the extension exists but not active
-            if (this._vscodeWrapper.azureAccountExtension) {
-                await this._vscodeWrapper.azureAccountExtension.activate();
-                return this.handleExtensionActivation();
-            } else {
-                // Show recommendation to download the azure account extension
-                const selection = await this._vscodeWrapper.showInformationMessage(LocalizedConstants.msgPromptRetryFirewallRuleExtNotInstalled,
-                    LocalizedConstants.downloadAndInstallLabel);
-                if (selection === LocalizedConstants.downloadAndInstallLabel) {
-                    await this._vscodeWrapper.executeCommand(Constants.cmdOpenExtension, Constants.azureAccountExtensionId);
-                    this._vscodeWrapper.onDidChangeExtensions(async (e) => {
-                        // Activate the Azure Account extension and call the function again
-                        if (this._vscodeWrapper.azureAccountExtension) {
-                            await this._vscodeWrapper.azureAccountExtension.activate();
-                            await this.handleExtensionActivation();
-                        }
-                    });
-                }
-                return false;
+            // If no match or no accountId present, need to add an azure account
+            let selection = await this._vscodeWrapper.showInformationMessage('Connection failed - firewall rule.', 'Add Azure Account');
+            if (selection === 'Add Azure Account') {
```

Comment:
We need to use string constants here. For example, if someone's using a localized version of vscode-mssql, this would fail.